ipa-core/src/protocol/ipa_prf/shuffle/malicious.rs

```diff
@@ -223,16 +323,23 @@ async fn h3_verify<C: Context, S: BooleanArray>(
 ///
 /// ## Panics
 /// Panics when conversion from `BooleanArray` to `Vec<Gf32Bit` fails.
-fn compute_row_hash<S, I>(keys: &[StdArray<Gf32Bit, 1>], row_iterator: I) -> Hash
+fn compute_row_hash<S, B, I>(keys: &[StdArray<Gf32Bit, 1>], row_iterator: I) -> Hash
```

Comment:
Getting a `StdArray` is an artifact of using the low-level reveal directly. I think it would be better to do `Gf32Bit::from_array` immediately upon reveal and work with regular values, rather than pass the array type around to a bunch of other code.

I also think it would be worth adding a comment at the call to `malicious_reveal` explaining why that is done here (because shuffle is a bespoke malicious protocol thus can't rely on the context to select appropriately between semi-honest and malicious reveal).